packages/perseus/src/article-renderer.tsx

```diff
@@ -162,9 +163,32 @@ class ArticleRenderer
     };
 
     _sections: () => any = () => {
-        return Array.isArray(this.props.json)
+        const sections = Array.isArray(this.props.json)
             ? this.props.json
             : [this.props.json];
+
+        // In JIPT context we split sections to paragraphs in order match the
+        // translatable strings found on Crowdin when rendering articles in
+        // the WYSIWYG mode for translation. This is needed for the jipt.js
+        // integration in order to attribute the rendered strings on Crowdin.
+        if (getDependencies().JIPT.useJIPT) {
+            const paragraphs: Array<PerseusRenderer> = [];
+
+            for (const section of sections) {
+                JiptParagraphs.parseToArray(section.content).forEach(
+                    (paragraph) => {
+                        paragraphs.push({
+                            ...section,
+                            content: paragraph,
+                        });
+                    },
+                );
+            }
+
+            return paragraphs;
```

Comment:
One thought I had. This logic is now in 2 (or is it 3) places between webapp and Perseus.  Would it make sense to add a function onto the `JIPT` dependency object that does this work?  Then at least we'd just call that function from this location and the function could live in webapp closer to the other parsing/splitting code from Khan/webapp#19917. Just a thought.

Reply:
Good idea, this can definitely be captured in a higher level utility function.